src/server.ts

```diff
@@ -44,17 +45,13 @@ async function handleRequest(
   root: string,
   options: WebServerOptions
 ) {
-  const pathParts = req.url?.split('/') || [];
+  const url = new URL(req.url || '/', `http://localhost:${options.port}`);
+  const pathParts = url.pathname.split('/');
   const originalPath = path.join(root, ...pathParts);
-  if (req.url?.endsWith('/')) {
-    pathParts.push('index.html');
+  if (pathParts[pathParts.length - 1] == '') {
```

Comment:
Roses are red
Violets are blue
Use triple equals
Because "1" == true

Reply:
Violets are blue
Roses are red
Javascript sucks
What more can be said?

Reply:
At perf time, it oft should be noted:
A Googler, inspired, once quoted
In a PR description
A poetic conniption
And so to Eng Manager was promoted

Reply:
YES THAT'S THE ONE